sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -135,11 +135,21 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 				skl_hda_be_dai_links[IDISP_DAI_COUNT +
 					HDAC_DAI_COUNT + i];
 		}
-	} else if (codec_count == 2 && codec_mask & IDISP_CODEC_MASK) {
+	} else if (codec_count <= 2) {
 		num_links = ARRAY_SIZE(skl_hda_be_dai_links);
 		num_route = ARRAY_SIZE(skl_hda_map);
 		card->dapm_widgets = skl_hda_widgets;
 		card->num_dapm_widgets = ARRAY_SIZE(skl_hda_widgets);
+		if (!(codec_mask & IDISP_CODEC_MASK)) {
+			if (codec_count == 2)
+				return -EINVAL;
+			for (i = 0; i < IDISP_DAI_COUNT; i++) {
+				skl_hda_be_dai_links[i].codecs->name =
+					"snd-soc-dummy";
+				skl_hda_be_dai_links[i].codecs->dai_name =
+					"snd-soc-dummy-dai";
```

Comment:
If I understand correctly, `skl_hda_be_dai_links[i].codecs` is actually an array of `skl_hda_be_dai_links[i].num_codecs` elements. So, the above actually means `skl_hda_be_dai_links[i].codecs[0]` and it assumes that `.num_codecs` is at least 1. Maybe indeed this is a valid assumption, but it would be better to use an array representation instead of just a pointer and maybe add a comment, saying that array is guaranteed to have at least one element in it?

Reply:
@lyakh Ack. I changed this for the "production version" and used the common macros instead.